bot/exts/evergreen/connect_four.py

```diff
@@ -0,0 +1,440 @@
+            channel: discord.TextChannel,
+            player1: discord.Member,
+            player2: typing.Optional[discord.Member],
+            size: int = 7,
+    ) -> None:
+
+        self.bot = bot
+        self.channel = channel
+        self.player1 = player1
+        self.player2 = player2 or AI(self.bot, game=self)
+
+        self.grid = self.generate_board(size)
+        self.grid_size = size
+
+        self.unicode_numbers = UNICODE_NUMBERS[:self.grid_size]
+
+        self.message = None
+
+        self.player_active = None
+        self.player_inactive = None
+
+    @staticmethod
+    def generate_board(size: int) -> typing.List[typing.List[int]]:
+        """Generate the connect 4 board."""
+        return [[0 for _ in range(size)] for _ in range(size)]
+
+    async def print_grid(self) -> None:
+        """Formats and outputs the Connect Four grid to the channel."""
+        title = (
+            f'Connect 4: {self.player1.display_name}'
+            f' VS {self.bot.user.display_name if isinstance(self.player2, AI) else self.player2.display_name}'
+        )
+
+        rows = [" ".join(EMOJIS[s] for s in row) for row in self.grid]
+        first_row = " ".join(x for x in NUMBERS[:self.grid_size])
+        formatted_grid = "\n".join([first_row] + rows)
+        embed = discord.Embed(title=title, description=formatted_grid)
+
+        if self.message:
+            await self.message.edit(embed=embed)
+        else:
+            self.message = await self.channel.send(content='Loading ....')
+            for emoji in self.unicode_numbers:
+                await self.message.add_reaction(emoji)
+            await self.message.add_reaction(CROSS_EMOJI)
+            await self.message.edit(content=None, embed=embed)
+
+    async def game_over(self, winner: discord.user, loser: discord.user) -> None:
+        """Removes games from list of current games and announces to public chat."""
+        await self.channel.send(f"Game Over! {winner.mention} won against {loser.mention}")
+        await self.print_grid()
+
+    async def start_game(self) -> None:
+        """Begins the game."""
+        self.player_active, self.player_inactive = self.player1, self.player2
+
+        while True:
+            await self.print_grid()
+
+            if isinstance(self.player_active, AI):
+                coords = self.player_active.play()
+                if not coords:
+                    await self.channel.send("Game Over! It's A Draw :tada:")
+                    await self.print_grid()
+            else:
+                coords = await self.player_turn()
+
+            if not coords:
+                return
+
+            if self.check_win(coords, 1 if self.player_active == self.player1 else 2):
+                await self.game_over(
+                    self.bot.user if isinstance(self.player_active, AI) else {self.player_active},
+                    self.bot.user if isinstance(self.player_inactive, AI) else {self.player_inactive},
```

Comment:
These two lines can cause an error, as the function expects two discord members, but gets sets instead. Try removing the braces here.

Reply:
Ah yes.